src/main/groovy/com/deploygate/gradle/plugins/artifacts/ApkInfoCompat.groovy

```diff
@@ -0,0 +1,136 @@
+package com.deploygate.gradle.plugins.artifacts
+
+class ApkInfoCompat {
+    private ApkInfoCompat() {
+    }
+
+    static ApkInfo blank(String name) {
+        return new ApkInfo() {
+            @Override
+            String getVariantName() {
+                return name
+            }
+
+            @Override
+            File getApkFile() {
+                return null
+            }
+
+            @Override
+            boolean isSigningReady() {
+                return true
+            }
+
+            @Override
+            boolean isUniversalApk() {
+                return true
+            }
+        }
+    }
+
+    static ApkInfo from(/*ApplicationVariant*/ applicationVariant, /*BaseVariantOutput*/ variantOutput) {
```

Comment:
Should we remove argument type declarations where we are expecting them?

Reply:
Yes. In the first place, these types cannot be resolved because we don't have any dependencies for the Android build tools.
Of course, we have a choice to embed the dependency but it doesn't sound realistic to me. 
Because the dependent APIs we are using are not robust even though type names are not changed.

As you know, 2.x release version, 3.0.0-preview and 3.0.0 release version have different APIs and we are addressing them by different approaches. 🤷‍♂️ 
Therefore, we shouldn't embed the dependency to this plugin. And then, we need to solve types dynamically.

Reply:
I see, sounds good!